### PR TITLE
change on kafka and update documentation

### PR DESCRIPTION
**Pull request by SergioLangaritaBenitez**

### CODE REVIEW

File: docpage/docs/02.-Inputs/02.1.-dcache.md
Comment (esparig) [inline docpage/docs/02.-Inputs/02.1.-dcache.md:1]: Please maintain the same structure throughout the documentation; giving a definition at the beginning of the section could be a good idea (see comments on 2.3-5).

File: docpage/docs/02.-Inputs/02.2.-generic.md
Comment (esparig) [inline docpage/docs/02.-Inputs/02.2.-generic.md:1]: Please maintain the same structure throughout the documentation; giving a definition at the beginning of the section could be a good idea (see comments on 2.3-5).

File: docpage/docs/03.-Outputs/03.1.-OSCAR.md
Comment (esparig) [inline docpage/docs/03.-Outputs/03.1.-OSCAR.md:1]: Please maintain the same structure throughout the documentation; giving a definition at the beginning of the section could be a good idea (see comments on 2.3-5).

**Review by esparig — CHANGES_REQUESTED** [review]
Good work on this PR! I've left some suggestions for consideration. Feel free to discuss or implement as you see fit.

**Comment by SergioLangaritaBenitez** [comment]
I changed the code and adapted it to the Python style guide (PEP 8). Also, I changed the documentation as it was required.

File: cli/dcnios-cli.py [hunk 1]
```diff
@@ -19,175 +19,199 @@
...
+                    dcachecontent = prepareforAll(dcachefile)
+                    nifi.create(dcache["name"], dcachecontent)
+                    command = "simple-client.py --state /state/" \
+                        + dcache["statefile"] + " --endpoint " \
+                        + dcache["endpoint"] \
+                        + " --user "+dcache["user"] + " --password " \
+                        + dcache["password"] + " " + dcache["folder"]
+                    nifi.changeVariable(dcache["name"], "command", command)
                     newProcessInfo(dcache["name"])
                     updateComponent(dcache)
```
Comment (esparig) [inline cli/dcnios-cli.py:104]: I would create an external function for this block of code and the same for the rest of the values (dCache, OSCAR, generic...). The resulting code would be easier to read and maintain.

File: docpage/docs/02.-Inputs/Kafka.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ - The group identifier indicating the consumer group. Required.
+ - In case the Kafka topics are separated by `key:value` set the argument `separate_by_key` as true and select the demarcator with `message_demarcator`:
+    - separate_by_key
+    - message_demarcator
+ - The kafka recipe in [IM](https://www.grycap.upv.es/im/index.php) support the SASL_SSL security protocol. So, the user and password of the SASL_SSL security protocol must be set:
+    - sasl_username: `user`
+    - sasl_password: `password`
+
+
+The connection between NiFi and Kafka employs SSL context,the SSL Protocol used by default is TLS, and the Truststore Type is PKCS12. SSL context only needs the Truststore filename and Truststore password. The Truststore file must be under "/opt/nifi/nifi-current/data/" directory.
```
Comment (esparig) [inline docpage/docs/02.-Inputs/Kafka.md:19]: Please, put a blank space after the comma.

File: docpage/docs/02.-Inputs/SQS.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
+---
+sidebar_position: 5
+---
 # SQS
 
-
+SQS is the ProcessGroup that consumes an AWS SQS queue , is comprised of:
```
Comment (esparig) [inline docpage/docs/02.-Inputs/SQS.md:6]: Please remove the extra blank space before the comma.

File: docpage/docs/02.-Inputs/dcache.md [hunk 4]
```diff
@@ -1,7 +1,9 @@
+---
+sidebar_position: 2
+---
 # dCache
-The main goal of this ProcessGroup is to get the events from dcache.
 
-It is necessary:
+dCache is the ProcessGroup that keep listen into dCache instance, is comprised of:
```
Comment (esparig) [inline docpage/docs/02.-Inputs/dcache.md:6]: How about?: "dCache is the ProcessGroup that monitors dCache instances for events"

File: docpage/docs/02.-Inputs/dcache.md [hunk 5]
```diff
@@ -1,7 +1,9 @@
...
 - An identifier name of the process. It must be unique.
 - Endpoint, user and password
```
Comment (esparig) [inline docpage/docs/02.-Inputs/dcache.md:8]: Use Oxford comma here: "Endpoint, user, and password."

File: docpage/docs/02.-Inputs/dcache.md [hunk 6]
```diff
@@ -1,7 +1,9 @@
...
 - Folder where active listening
```
Comment (esparig) [inline docpage/docs/02.-Inputs/dcache.md:9]: Folder to be monitored

File: docpage/docs/02.-Inputs/generic.md [hunk 7]
```diff
@@ -1,9 +1,11 @@
...
+---
 # Generic
 
-The generic section is made to deploy your ProcessGroups.
-Need to be defined:
-- The identifier name
-- The path of your processGroup (.json file)
+Generic section creates a ProcessGroup by given a ProcessGroup file, is comprised of:
+- An identifier name of the process. It must be unique.
+- The path of your ProcessGroup (.json file)
```
Comment (esparig) [inline docpage/docs/02.-Inputs/generic.md:8]: Could you explain to me what you mean by this paragraph?

File: docpage/docs/AWS.md [hunk 8]
```diff
@@ -1,12 +1,23 @@
+---
+sidebar_position: 4
+---
 # AWS
 
-DCNiOS can use some AWS as input. In all those cases it is necesary a valid pair of AWS Access Key and AWS Secret Key.
+> :warning: **DCNiOS can use some AWS services**: This section clarifies the configuration required to use the credentials!
```
Comment (esparig) [inline docpage/docs/AWS.md:6]: I don't think this should be a warning.

File: docpage/docs/AWS.md [hunk 9]
```diff
@@ -1,12 +1,23 @@
...
+    aws_access_key_id = AK<>
+    aws_secret_access_key = <>
+    ```
 - If DCNiOS does not successfully use the credentials as environment variables or in the credential file.
 DCNiOS gets them from configuration files named `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.
 This implementation is made to minimize the times that the credentials are written in the configuration file.
 
-AWS_DEFAULT_REGION is mandatory in any source that uses AWS of the configuration file.
-
+AWS_DEFAULT_REGION is mandatory in any source that uses AWS of the configuration file. Right now, the ProcessGroup that uses AWS credentials are:
```
Comment (esparig) [inline docpage/docs/AWS.md:21]: Please, we need to review the documentation texts out of the PR.

**Pull request merged**